pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -765,31 +766,41 @@ private void handleSubquery(Expression expression, long requestId, JsonNode json
    * - Case-insensitive cluster
    * - Table name in the format of [database_name].[table_name]
    *
-   * Drop the database part if there is no existing table in the format of [database_name].[table_name], but only
-   * [table_name].
+   * If the PinotConfiguration does not allow dots in table name, will do the below:
+   * Drop the prefix part for the format of [database_name].[table_name], keep only [table_name].
+   * If the PinotConfiguration allows dots in table name, we will not do the split
+   * @param tableName the table name in the query
+   * @param tableCache the table case-sensitive cache
+   * @param routingManager the routing mananger for testing whether a route exists
+   * @param config the Pinot Configuration
+   * @return
    */
-  private String getActualTableName(String tableName) {
+  @VisibleForTesting
+  static String getActualTableName(String tableName, TableCache tableCache, BrokerRoutingManager routingManager,
```

Comment:
We might not want to change this method. Even if we allow table name with database, we should probably still support the case of table name without database.

Reply:
Right now the code change has only one effect:
If we allow dots in table name, we will avoid striping the `[database_name].` from the table name. This will be needed in our case, that we wish to treat `[database_name].[table_name]` as the whole actual table name.

This behavior is under the feature flag so will be backwards compatible.

Reply:
What I meant is that the existing code should already be able to handle the `[database_name].[table_name]` as the table name (line 773, 801, 812 in the original code). We don't need to introduce the extra logic of reading properties from the config.

Reply:
I see. I did not expect the behavior to be the same but you are right.
I am still keeping the unit tests since they will ensure our expected behavior is not broken